### ISSUE TITLE
comparing results from multiple scenarios in a Jupyter Notebook

### ISSUE THREAD

**Issue by nesnoj** [windnode/WindNODE_ABW#64]
Notes
- In the current version there're many plots showing all scenarios which makes it somehow hard to grasp. For some plots we should pick out only relevant scenarios.
- Implement according to priority (1=high/must-have, 2=medium, 3=low)
- Unless otherwise specified, spatial resolution is ABW

## 0 Results Overview

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 0.1 | Total costs in EUR, LCOE in EUR/MWh, LCOH in EUR/MWh, Autonomy in %, CO2 Emissions in t CO2-eq., Land-use wind % (in relation to s1000f0 (sq)), Land-use PV in % (in relation to HS 1-perc. agri), Activation of battery storages (i.e. total cycles), Activation of heat storages (i.e. total cycles), Activation of DSM (i.e. total shifted energy) | see left | [PairGrid dot plot](https://seaborn.pydata.org/examples/pairgrid_dotplot.html) | Ziel ist es einen Überblick mittels der Kernkenngrößen zu allen Szenarien zu bekommen. scenarios on y-axis and data (key indicators) on the x-axis | 1 | X |

### 1.1 Installed Capacities Electricity/Heat

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 1.1.1 | Installed Capacity Electricity | MW | barplot stacked (by tech) | x: scenarios | 1 | X | 
| 1.1.2 | Installed Capacity Heat | MW | barplot stacked (by tech) | x: scenarios | 1 | X | 
| 1.1.3 | Installed Capacity Wind+PV (PV=ground+roof) | - | dotplot | x: wind, y: PV, maybe good to quickly grasp the major tech | 3 |  |

1.1.1: If el+th is too much in one plot, split into 2 plots.

### 1.2 Electricity and Heat Generation

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 1.2.1 | Generation per Technology | GWh | barplot stacked (by tech), grouped: el+th per scn in on group | x: scenarios | 1 | x |

1.2.1: If el+th is too much in one plot, split into 2 plots.

### 1.3 Electricity and Heat Demand

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 1.3.1 | Annual Demand | GWh | barplot stacked (by tech (el), sector (th)), grouped: el+th per scn in on group | x: scenarios, th: sum up efh+mfh | 1 | x |

1.3.1: If el+th is too much in one plot, split into 2 plots.

### 2.1 Available areas: Scenarios

Overview / cumulative sums:

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 2.1.0.1 | Potential Areas | GW, % (2 axes) | barplot grouped | similar to plot `Relative Required Area` in single scenario jupy nb but with abs. numbers -> abs. in GW, rel. in % related to legal SQ areas | 1 |  |

More detailed analysis in 2.1.1-2.1.3 .

#### 2.1.1 Available areas: Wind Energy

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 2.1.1.1 | Potential Areas Wind | - | geoplot | All avail. areas (geometries) plotted on mun shapes (s1000f0, s1000f1, s500f0, s500f1), geoms are avail. in DF `region.pot_areas_wec`, please use 'sq' for s1000f0 | 2 |  |
| 2.1.1.2 | Potential Areas Wind per mun | ha | barplot grouped: (s1000f0, s1000f1, s500f0, s500f1) per scn in on group | x: mun , geoms are avail. in DF `region.pot_areas_wec`, please use 'sq' for s1000f0 | 1 |  |

#### 2.1.2 Available areas: Ground-Mounted PV

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 2.1.2.1 | Potential Areas PV ground: railways | ha | geoplot | All avail. areas (geometries) plotted on mun shapes side by side (H, HS), geoms are avail. in DF `region.pot_areas_pv` | 2 |  |
| 2.1.2.2 | Potential Areas PV ground: motorways | ha | geoplot | All avail. areas (geometries) plotted on mun shapes side by side (H, HS), geoms are avail. in DF `region.pot_areas_pv` | 2 |  |
| 2.1.2.3 | Potential Areas PV ground: agricultural | ha | geoplot | All avail. areas (geometries) plotted on mun shapes side by side (H, HS), geoms are avail. in DF `region.pot_areas_pv` | 2 |  |
| 2.1.2.4 | Potential Areas PV ground: area sums | ha |  barplot grouped (railway, motorway, agri per scn in on group) | x: scenarios | 1 |  |

#### 2.1.3 Available areas: Roof-Mounted PV

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 2.1.3.1 | Potential Areas PV roof | ha | geoplot | Area sum Summe Flächen pro Gemeinde | 1 |  |

### 2.2 Area usage

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 2.2.1 | Rel. Area Usage Wind (cum. for ABW) | % | barplot grouped | For each scenario: rel. usage in relation to land use scenario of scenario, sorted by the 4 land use scenarios (see photo below) | 1 |  |
| 2.2.2 | Rel. Area Usage Wind (cum. for ABW) | % | barplot | Rel. usage in all scenarios in relation to the total region's area ([362900 ha](https://www.planungsregion-abw.de/index.php/raumbeobachtung/daten-und-fakten/)) | 1 |  |
| 2.2.3 | Rel. Area Usage PV ground (cum. for ABW) | % | barplot grouped | For each scenario: rel. usage in relation to land use scenario of scenario, sorted by the 6 land use scenarios (see photo below) | 1 |  |
| 2.2.4 | Rel. Area Usage PV ground (cum. for ABW) | % | barplot | Rel. usage in all scenarios in relation to the total region's area ([362900 ha](https://www.planungsregion-abw.de/index.php/raumbeobachtung/daten-und-fakten/)) | 1 |  |
| 2.2.5 | Rel. Area Usage PV roof (cum. for ABW) | % | barplot | Rel. usage in all scenarios | 1 |  |

Example Wind 2.2.1+2.2.2 (PV ground 2.2.3+2.2.4 analogously):
![plots1](https://user-images.githubusercontent.com/15910173/89951937-9a159980-dc2c-11ea-826c-31bdb11a2c14.jpg)

### 3.1 Energy Exchange

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 3.1.1 | Import and Export (National Grid) | GWh | barplot grouped | x: scenarios, **could be merged into 3.2.1?** | 1 |  |
| 3.1.2 | Intra-Regional Import and Export | GWh | barplot grouped | x: scenarios | 1 |  |

### 3.2 Autarky

(#85 needs to be done first)

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 3.2.1 | Demand+supply | GWh, % | barplot grouped | like plot `Electrical Autarky per Municipality (Energy Balance) and total region` in single scenario jupy nb but with scenarios instead of muns  | 1 |  |
| 3.2.2 | Autarky: energy + time | % | barplot grouped | time: 2 options: 1) mean of all time step autarky values, 2) perc. of autark hours -> to be specified after #85 | 1 |  |
| 3.2.3 | Autarky: energy violin | % | violin plot | like plot `Relative Electrical Autarky per Municipality (Energy Balance)` in single scenario jupy nb but with scenarios instead of muns | 2 |  |

## 4 Energy Mix

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 4.1 | Electricity Generation and Demand | GWh | barplot stacked by tech(+) or sector(-) | like plot `Power Generation and Demand` in single scenario jupy nb but with scenarios instead of muns | 1 | x |
| 4.2 | Heat Generation and Demand | GWh | barplot stacked by tech(+) or sector(-) | like plot `Power Generation and Demand` in single scenario jupy nb but with scenarios instead of muns | 1 | x |
| 4.3 | Full-Load Hours Electricity | h | barplot grouped by tech (wind, pv ground, pv roof, bio, ...)  | like plot `Power Generation and Demand` in single scenario jupy nb but with scenarios instead of muns | 2 |  |

## 5 Emissions

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 5.1 | Emissions absolute (sum) | t CO2 eq. | barplot y-stacked by sector (el, th, stor, grid) | x: scenarios | 1 | x |
| 5.2 | Emissions relative | % | barplot stacked by sector (el, th, stor, grid), scaled to 100% | x: scenarios | 3 |  |

## 6 Costs

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 6.1 | LCOE and LCOH | €/MWh | barplot grouped per scn with el+th | x: scenarios | 1 |  |
| 6.2 | RES share and LCOE and LCOH | €/MWh | bar/scatter plot sorted along increasing RES share | x: scenarios | 1 | x |

## 7 Power Grid

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 7.1 | Line Loading | % | whisker box plot, like `DSM Activation` plot in single scenario jupy nb  | x: scenarios | 1 |  |

## 8 Flexibility

It'd be good to have an overview plot on the flex. to quickly see the configuration/character of each scenario. I propose either 8.0.1 or 8.0.2. Alternatively, we could move this to the scenario overview (maybe better).

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 8.0.1 | Flex. Character | €/MWh | PairGrid dot plot (as in 0) showing the flex. character of scenario (RES power, DSM, autarky, pth stor, batteries, eff. el., eff. th.) | x: scenarios | 2 |  |
| 8.0.2 | Flex. Character | €/MWh | [radar chart](https://plotly.com/python/radar-chart/) showing the flex. character of scenario (RES power, DSM, autarky, pth stor, batteries, eff. el., eff. th.), N charts we assign the scenarios to, e.g. BAThigh+BATlimit in one radar and so on | x: scenarios | 2 |  |

### 8.1 Thermal Storages

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 8.1.1 | Storage use | - | whisker box plot grouped by scn with dec+cen, like `DSM Activation` plot in single scenario jupy nb  | x: scenarios | 1 |  |

### 8.2 Batteries

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 8.2.1 | Storage use | - | whisker box plot grouped by scn with large+small, like `DSM Activation` plot in single scenario jupy nb  | x: scenarios | 1 |  |

### 8.3 DSM

|# | Größe | Einheit | Plot/Tabelle | Kommentar | Prio | Done? |
|-------|----------|-----------|-----------------|-----------------|---|---|
| 8.3.1 | Total Sum (Ratio DSM / demand hh) | % | bar plot | x: scenarios | 1 |  |

## 9 Energy Exchange

As proposed in https://github.com/windnode/WindNODE_ABW/issues/96#issuecomment-670477150 I integrated this section into 3.

**Comment by nesnoj** [comment]
I :heart:  the PairGrid dot plot very much, but not sure if this can be done in plotly ([probably it can](http://www.storytellingwithdata.com/blog/2018/8/21/august-swdchallenge-recap-the-dot-plot)).

**Comment by nesnoj** [comment]
- add note on PV home storage systems in NB, see https://github.com/windnode/WindNODE_ABW/issues/100#issuecomment-665555414

**Comment by nesnoj** [comment]
Ok, I adjusted the content above and added priorities, feel free to amend.
I'll prepare the next steps..

**Comment by nesnoj** [comment]
I added a first version, see #131.

Also, we gotta discuss the scenario order:
1. Do we want a constant order? would make it easier to catch
2. If 1: which order?

**Comment by gplssm** [comment]
Looking through the results, I saw some oddities. For example the scenario with most PV yields highest CO2 emissions. 
Furthermore, number aren't always comparable across scenarios (i.e. BAT with BAT+).

I would suggest to normalize some of the data shown in the PairGrid plot.

- [ ] Emissions -> Emissions / sum(generation) in g/kWh 
- [ ] El. storage -> Utilization Rate in %
- [ ] Heat storage -> Utilization Rate in %
- [ ] DSM ->?do we have relative numbers here?

@nesnoj if you agree, @nailend can start here tomorrow morning.

**Comment by nesnoj** [comment]
Yes, I like the idea of having rel. numbers!

> @nesnoj if you agree, @nailend can start here tomorrow morning.

The function in question is a little naughty but if you dare, @nailend, go for it :)

> * DSM ->?do we have relative numbers here?

Yes, a Relative Utilization was lately introduced in single nb 8.3.4, this seems appropriate. We have increase+decrease over there, should we simply go for a mean of both?!

**Comment by gplssm** [comment]
> The function in question is a little naughty but if you dare, @nailend, go for it :)

But it's not too bad. And honestly, it's mostly about calculating some numbers [here](https://github.com/windnode/WindNODE_ABW/blob/ffc85d8a5e75cafc6d40c3f3712dd2eca5c76d07/windnode_abw/tools/draw.py#L889-L911).
Could you please add calculated number there and not replace the old absolute numbers by relative numbers? I plan to return this data for later use.

> > * DSM ->?do we have relative numbers here?
> 
> Yes, a Relative Utilization was lately introduced in single nb 8.3.4, this seems appropriate. We have increase+decrease over there, should we simply go for a mean of both?!

We should take only take decrease since the potential for decreasing load is smaller and thus it defines the overall limit.

**Comment by nesnoj** [comment]
> But it's not too bad. And honestly, it's mostly about calculating some numbers [here](https://github.com/windnode/WindNODE_ABW/blob/ffc85d8a5e75cafc6d40c3f3712dd2eca5c76d07/windnode_abw/tools/draw.py#L889-L911).

Yes, was only some kind of criticism on the bad coding style ;)
Go ahead :+1:

**Comment by nailend** [comment]
> We should take only take decrease since the potential for decreasing load is smaller and thus it defines the overall limit.

I just discovered that we 'increase' was used for Net DSM activation?! @gplssm should I correct this?
https://github.com/windnode/WindNODE_ABW/blob/196a2549920692dd63308f810b97038b6398dfb1/windnode_abw/analysis/tools.py#L1379

**Comment by gplssm** [comment]
> > We should take only take decrease since the potential for decreasing load is smaller and thus it defines the overall limit.
> 
> I just discovered that we 'increase' was used for Net DSM activation?! @gplssm should I correct this?
> 
> https://github.com/windnode/WindNODE_ABW/blob/196a2549920692dd63308f810b97038b6398dfb1/windnode_abw/analysis/tools.py#L1379

I'm aware of that. In which sense you want to correct it? Taking decrease instead increase because the potential for decrease is expected to be lower? If so, it wouldn't matter, because in the end both are balanced.

**Comment by nailend** [comment]
true, forgot about that.
